app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -746,18 +758,48 @@ public boolean onBackPressed() {
             return true;
         }
 
+        if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_CLOSE) {
+            if (!isPlayerAvailable() || player.videoPlayerSelected()) {
+                bottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+            } else {
+                // Since it's not a video playing we can't close the player
+                bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+            }
+            return true;
+        }
+
+        if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_HIDE) {
+            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+            return true;
+        }
+
         // If we have something in history of played items we replay it here
         if (isPlayerAvailable()
                 && player.getPlayQueue() != null
                 && player.videoPlayerSelected()
                 && player.getPlayQueue().previous()) {
             return true;
         }
+
+        if (behavior == PlayerHelper.BackBehavior.BACK_BEHAVIOR_PREV_PLAYLIST) {
+            bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
+            return true;
+        }
```

Comment:
Could you explain how this code achieves the behavior you described? I see nothing checking whether the two videos belong to the same playlist
> open previous video only in case it was playing from the same playlist as current one

Reply:
Explained now